### PR TITLE
`Projections, Extents, and Layout Definitions` doc

### PR DESCRIPTION
**Pull request by fosskers**
*The other day, on gitter:*

Me: Alright, I'm digging into the relationship between projections, layout definitions, and extents.

@echeipesh : That sounds like a title of an excellent doc page I wish we had.

### CODE REVIEW

**Comment by fosskers** [comment]
Not confident of the line: 

> "min" and "max" here are CRS specific, as the location of the point `(0,0)` varies between different CRS.

**Comment by lossyrob** [comment]
Would be useful to comment on the diff line you're referring to.

File: docs/concepts/projections-extents-layouts.md [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+`geotrellis.raster.{ GridExtent, RasterExtent }`,
+`geotrellis.raster.reproject.ReprojectRasterExtent`
+
+**What is an extent?**
+
+An `Extent` is a rectangular section of a 2D projection of the Earth. It is
+represented by two coordinate pairs that are its "min" and "max" corners in
+some Coorindate Reference System. "min" and "max" here are CRS
+specific, as the location of the point `(0,0)` varies between different CRS.
+An Extent can also be referred to as a *Bounding Box*.
```
Comment (lossyrob) [inline docs/concepts/projections-extents-layouts.md]: The whole extent is CRS specific; anything relating to the Extent will have to be in that CRS, and users of the extent should know the CRS of the extent if they are trying to combine it with other things, either by knowing implicitly that all the geometries they are working with are in the same CRS, or explicitly by holding onto CRS instances, and reprojecting where necessary.

**Comment by fosskers** [comment]
https://github.com/geotrellis/geotrellis/pull/1608/files#diff-345d6013e8e8878b87ef9caff9208ccaR78

File: docs/concepts/projections-extents-layouts.md (hunk 1 above)
Comment (fosskers) [inline docs/concepts/projections-extents-layouts.md]: Yup, that's mentioned in the next paragraph.

**Pull request merged**